src/main/java/com/dataguise/saas/service/DetectionTaskService.java

```diff
@@ -0,0 +1,192 @@
+import io.swagger.client.model.DgDiscoverTask;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class DetectionTaskService {
+
+    private static Logger logger = Logger.getLogger(DetectionTaskService.class);
+
+    @Autowired private DgSecureRestDriver restDriver;
+    @Autowired private SettingsController settingsController;
+    @Autowired private TaskBuilderService taskBuilderService;
+
+
+    /**
+     * Create a detection task
+     * @param detectionTask
+     * @return
+     * @throws BadGatewayException
+     */
+    public Integer createTask(@RequestBody DgDiscoverTask detectionTask, String sourceModule) throws BadGatewayException {
+        Integer taskId = null;
+        try {
+            taskId = restDriver.createDetectionTask(detectionTask, sourceModule);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Detection task created with id: " + taskId);
+        return taskId;
+    }
+
+    /**
+     * Edit a detection task
+     * @param id
+     * @param detectionTask
+     * @return
+     * @throws BadGatewayException
+     * @throws BadRequestException
+     */
+    public Integer editTask(@PathVariable Integer id, @RequestBody DgDiscoverTask detectionTask)
+            throws BadGatewayException, BadRequestException {
+        Integer taskId = null;
+        if( id == null) {
+            throw new BadRequestException("Task " + ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage());
+        }
+        try {
+            taskId = restDriver.editDetectionTask(id, detectionTask);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Detection task edited with id: " + taskId);
+        return taskId;
+    }
+
+    /**
+     * Execute a detection task
+     * @param taskId
+     * @return
+     * @throws BadRequestException
+     * @throws BadGatewayException
+     */
+    public Integer executeTask(@PathVariable Integer taskId) throws BadRequestException, BadGatewayException {
+        if( taskId == null) {
+            throw new BadRequestException("Task " + ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage());
+        }
+        Integer taskInstanceId = null;
+        try {
+            taskInstanceId = restDriver.executeDetectionTask(taskId);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Detection task executed with instance id: " + taskInstanceId);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/DetectionTaskService.java L92
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1249937482>*